src/plonk/lookup/prover.rs

```diff
@@ -414,16 +432,25 @@ impl<'a, C: CurveAffine> Committed<C> {
     ) {
         let permuted = self.permuted;
 
+        let active_rows = Polynomial::one_minus(pk.l_last.clone() + &pk.l_blind);
+
         let expressions = iter::empty()
-            // l_0(X) * (1 - z'(X)) = 0
+            // l_0(X) * (1 - z(X)) = 0
             .chain(Some(
                 Polynomial::one_minus(self.product_coset.clone()) * &pk.l0,
             ))
-            // z'(X) (a'(X) + \beta) (s'(X) + \gamma)
-            // - z'(\omega^{-1} X) (\theta^{m-1} a_0(X) + ... + a_{m-1}(X) + \beta) (\theta^{m-1} s_0(X) + ... + s_{m-1}(X) + \gamma)
+            // l_last(X) * (z(X)^2 - z(X)) = 0
+            .chain(Some(
+                (self.product_coset.clone() * &self.product_coset - &self.product_coset)
+                    * &pk.l_last,
```

Comment:
These expressions are not encoded into the `PinnedVerificationKey` because there aren't many degrees of freedom for them. We could reorder these to be consistent with the book; the reason it's not done here is because it would require cloning `pk.l_last` in the current API.